src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/sdk/Sdk.props

```diff
@@ -11,7 +11,12 @@ Copyright (c) .NET Foundation. All rights reserved.
 -->
 <Project>
 
+  <!-- Set a flag to indicate that aot packages are loaded via SDK  -->
+  <PropertyGroup>
+    <AotRuntimePackageLoadedViaSDK Condition="'$(ILCompilerTargetsPath)'== ''">true</AotRuntimePackageLoadedViaSDK>
+  </PropertyGroup>
+
   <!-- Only import the build props if the NativeAot package isn't referenced via NuGet. -->
-  <Import Project="$(MSBuildThisFileDirectory)..\build\Microsoft.DotNet.ILCompiler.targets" Condition="'$(PublishAot)' == 'true'"/>
+  <Import Project="$(MSBuildThisFileDirectory)..\build\Microsoft.DotNet.ILCompiler.props" Condition="'$(PublishAot)' == 'true' and '$(ILCompilerTargetsPath)' == ''" />
```

Comment:
I wasn't sure if there is any way that the SDK path can be activated before the explicit ILCompiler package reference can set the sentinel since there are non-PublishAOT calls in the [SDK](https://github.com/dotnet/sdk/pull/26652/files#diff-cde7a8c0e370b40890bd6539212594fdbb8688a55f9f40524d535c3972dd54a6R156) that might set the sentinel. 

So there are two concerns here,

- If explicit `ILCompiler` package references always set the sentinel first, then its ok to remove the check here
- There needs to be a guard for non-PublishAot SDK code paths to not import aot targets since [Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props](https://github.com/dotnet/sdk/pull/26652/files#diff-cde7a8c0e370b40890bd6539212594fdbb8688a55f9f40524d535c3972dd54a6) and [Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets](https://github.com/dotnet/sdk/pull/26652/files#diff-b8dad9da3662ff7878709ecc6bba9f3249c3dd1471b3e67ae13ee6d009604040) might get invoked in non-PublishAot scenarios. I also added a check in the targets file before importing the native AOT target file and if the above concern is not valid, we can remove it from the props file